kaiax/gasless/README.md

```diff
@@ -1,32 +1,70 @@
 # kaiax/gasless
 
-This module is responsible for enabling gasless transactions.
+This module is responsible for facilitating gasless transactions specified by [KIP-247](https://kips.kaia.io/KIPs/kip-247).
 
 ## Concepts
 
-### Transaction patterns
+Gasless transaction (GaslessTx) consists of two types: gasless approve transaction (GaslessApproveTX), and gasless swap transaction (GaslessSwapTx).
 
-- LendTx
-- GaslessApproveTx
-- GaslessSwapTx
+Note that gasless transaction does not mean the gas prices of gasless transactions are zero, but it means that proposer will lend the user with gas fee and user will pay back during gasless swap.
+
+The gas fee of gasless transaction's is funded by block proposer (i.e., lend transaction generated by `GetLendTxGenerator`), and user repays the lent amount during gasless swap.
 
 ### Transaction pool rules
 
-TBU
+#### Ready
+
+This module is responsible for promoting gasless transactions.
+Sender's nonce of GaslessSwapTx is checked to distinguish if GaslessApproveTx is expected. If `tx.nonce == GetNonce(sender) + 1`, GaslessApproveTx is expected. If `tx.nonce == GetNonce(sender)`, GaslessApproveTx is not expected.
+
+If GaslessApproveTx is expected, GaslessApproveTx and GaslessSwapTx can be promoted when they are both ready for execution.
+Otherwise, GaslessSwapTx can be promoted when it is ready for execution.
+
+See ready condition [KIP-247](https://kips.kaia.io/KIPs/kip-247) and the implementation `IsExecutable(approveTxOrNil, swapTx *types.Transaction) bool`.
```

Comment:
I can add the return value of IsExecutable and debug_IsGaslessTx after this PR is merged, but will you do it in this PR?

Reply:
added 6426d4b